src/test/java/io/cdap/plugin/WindowsAggregationUtilTest.java

```diff
@@ -145,49 +149,92 @@ public void setUp() {
-         config, schema, schema);
-       Assert.assertNotNull(result);
-     } catch (Exception e) {
-       Assert.fail("Exception not expected");
-     }
+  public void testTransformForRange() {
+    try {
+      if (mocked == null || mocked.isClosed()) {
+        mocked = Mockito.mockConstruction(SQLContext.class,
+          (mock, context) -> {
+            Mockito.when(mock.createDataFrame(javaRDDRow, structSchema)).thenReturn(data);
+            Mockito.when(mock.udf()).thenReturn(udfRegistration);
+            mockStatic = Mockito.mockStatic(DataFrames.class);
+            mockStatic.when(DataFrames.toDataType(schema)).thenReturn(structSchema);
+          });
+      }
+      Mockito.when(config.getWindowFrameType()).thenReturn(WindowAggregationConfig.WindowFrameType.RANGE);
+      JavaRDD<StructuredRecord> result = WindowsAggregationUtil.transform(sparkExecutionPluginContext, javaRDDSR,
+                                                                          config, schema, schema);
+      Assert.assertNotNull(result);
+    } catch (Exception e) {
+      Assert.fail("Exception not expected");
+    } finally {
+      if (mockStatic != null && mocked != null) {
+        mockStatic.closeOnDemand();
+        mocked.closeOnDemand();
+      }
+    }
+  }
+
+  @Test
+  public void testTransformForRow() {
+    try {
+      if (mocked == null) {
+        mocked = Mockito.mockConstruction(SQLContext.class,
+          (mock, context) -> {
+            Mockito.when(mock.createDataFrame(javaRDDRow, structSchema)).thenReturn(data);
+            Mockito.when(mock.udf()).thenReturn(udfRegistration);
+            mockStatic = Mockito.mockStatic(DataFrames.class);
+            mockStatic.when(DataFrames.toDataType(schema)).thenReturn(structSchema);
+          });
+      }
+      Mockito.when(config.getWindowFrameType()).thenReturn(WindowAggregationConfig.WindowFrameType.ROW);
+      JavaRDD<StructuredRecord> result = WindowsAggregationUtil.transform(sparkExecutionPluginContext, javaRDDSR,
+                                                                          config, schema, schema);
+      Assert.assertNotNull(result);
+    } catch (Exception e) {
+      Assert.fail("Exception not expected");
+    } finally {
+      if (mockStatic != null && mocked != null) {
+        mockStatic.closeOnDemand();
+        mocked.closeOnDemand();
+      }
+    }
   }
 
   @Test(expected = InvalidParameterException.class)
   public void testInvalidParameter() {
-    try (MockedConstruction<SQLContext> mocked = Mockito.mockConstruction(SQLContext.class,
-      (mock, context) -> {
-        Mockito.when(mock.createDataFrame(javaRDDRow, structSchema)).thenReturn(data);
-        Mockito.when(mock.udf()).thenReturn(udfRegistration);
-        Mockito.mockStatic(DataFrames.class).when(DataFrames.toDataType(schema)).thenReturn(structSchema);
-      })) {
+    try {
+      if (mocked == null || mocked.isClosed()) {
```

Comment:
Could you describe the reason for adding these checks at the beginning and then closing the objects at the end?

Reply:
To avoid issue "Static mocking is already registered in the current thread" we need to close and assign only if mock is closed or null